tests/test_abacus_spin.py

```diff
@@ -0,0 +1,157 @@
+        mysys = dpdata.LabeledSystem("abacus.spin", fmt="abacus/scf")
+        data = mysys.data
+        self.assertAlmostEqual(data["energies"][0], -6818.719409466637)
+        np.testing.assert_almost_equal(
+            data["spins"][0],
+            [
+                [-0.0000002724, -0.0000001728, 2.4000001004],
+                [-0.0000003180, -0.0000002299, 2.3999994597],
+            ],
+            decimal=8,
+        )
+        np.testing.assert_almost_equal(
+            data["mag_forces"][0],
+            [
+                [-0.0000175013, -0.0000418680, -0.3669618965],
+                [-0.0000161517, -0.0000195198, -0.3669821632],
+            ],
+            decimal=8,
+        )
+
+        # dump to deepmd-npy
+        mysys.to(file_name=self.dump_path, fmt="deepmd/npy")
+        self.assertTrue(os.path.isfile(f"{self.dump_path}/set.000/spins.npy"))
+        self.assertTrue(os.path.isfile(f"{self.dump_path}/set.000/mag_forces.npy"))
+
+        sys2 = dpdata.LabeledSystem(self.dump_path, fmt="deepmd/npy")
+        np.testing.assert_almost_equal(data["spins"], sys2.data["spins"], decimal=8)
+        np.testing.assert_almost_equal(
+            data["mag_forces"], sys2.data["mag_forces"], decimal=8
+        )
+
+    def test_relax(self):
+        os.system("cp abacus.spin/INPUT.relax abacus.spin/INPUT")
+        mysys = dpdata.LabeledSystem("abacus.spin", fmt="abacus/relax")
+        data = mysys.data
+        spins_ref = np.array(
+            [
+                [
+                    [1.16909819, 1.16895965, 1.16895485],
+                    [1.16827825, 1.16832716, 1.16836899],
+                ],
+                [
+                    [1.25007143, 1.25006167, 1.25004587],
+                    [1.25015764, 1.2501678, 1.25018344],
+                ],
+                [
+                    [1.24984994, 1.24977108, 1.24978313],
+                    [1.24996533, 1.2500441, 1.25003208],
+                ],
+            ]
+        )
+        magforces_ref = np.array(
+            [
+                [[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]],
+                [
+                    [-0.16734626, -0.16735378, -0.16735617],
+                    [-0.16836467, -0.16835897, -0.16835625],
+                ],
+                [
+                    [-0.16573406, -0.16574627, -0.1657445],
+                    [-0.16619489, -0.16617948, -0.16618272],
+                ],
+            ]
+        )
+        self.assertEqual(len(data["spins"]), 3)
+        self.assertEqual(len(data["mag_forces"]), 3)
+        np.testing.assert_almost_equal(data["spins"], spins_ref, decimal=8)
+        np.testing.assert_almost_equal(data["mag_forces"], magforces_ref, decimal=8)
+
+        # dump to deepmd-npy
+        mysys.to(file_name=self.dump_path, fmt="deepmd/npy")
+        self.assertTrue(os.path.isfile(f"{self.dump_path}/set.000/spins.npy"))
+        self.assertTrue(os.path.isfile(f"{self.dump_path}/set.000/mag_forces.npy"))
```

Comment:
I think we use a different filename in the deepmd-kit

Reply:
Hi，@wanghan-iapcm ，I need to clarify what the name should be, and they should be the same in DeepMD, LAMMPS, and ABACUS. In previous disccussion, you suggest the name "spins" and approve name "mag_forces".

Reply:
We haven't had a documentation for spin data in the DeePMD-kit. xref: https://github.com/deepmodeling/deepmd-kit/issues/3760

Reply:
@pxlxingliang please follow PR #727 to support the deepmd-kit labeling files doced in https://github.com/deepmodeling/deepmd-kit/pull/3762